src/adapter/src/coord/timeline.rs

```diff
@@ -52,24 +52,21 @@ pub(crate) struct TimelineState<T> {
     pub(crate) read_holds: ReadHolds<T>,
 }
 
-/// A timeline is either currently writing or currently reading.
-///
-/// At each time, writes happen and then reads happen, meaning that writes at a time are
-/// visible to exactly reads at that time or greater, and no other times.
-enum TimestampOracleState<T> {
-    /// The timeline is producing collection updates timestamped with the argument.
-    Writing(T),
-    /// The timeline is observing collections aot the time of the argument.
-    Reading(T),
+/// A timeline can perform reads and writes. Reads happen at the read timestamp
+/// and writes happen at the write timestamp. After the write has completed,
+/// but before a response is sent, the read timestamp must be updated to the
+/// same value as the write timestamp.
```

Comment:
Is the condition maybe weaker, that "before responding to a write, the read timestamp must be updated to the timestamp of the write"? Or must the read timestamp be updated to `self.write_ts`?

Reply:
What about something like 
> After the write has completed, but before a response is sent, the read timestamp must be updated to a value greater than or equal to `self.write_ts`.

That's a bit weaker and more accurate.